### PR TITLE
feat(@aws-amplify/ui-components): Trim inputs before submit #7408

### PR DESCRIPTION
**Pull request by dennybiasiolli**
#### Description of changes
When user inputs have trailing or leading whitespace, Cognito returns

Value at 'username' failed to satisfy constraint: Member must satisfy regular expression pattern: [\p{L}\p{M}\p{S}\p{N}\p{P}]+
which is not readable to end users.
This pr improves #7408 trimming the `userInput` field also on `forgotPasswordSubmit` to avoid these errors.
Other than this, I added a few tests to make sure everything works as expected 😉 

#### Issue #, if available
#7408 and #6737

#### Description of how you validated changes
Tests and local dev env.

#### Checklist
- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/components/amplify-forgot-password/amplify-forgot-password.tsx [hunk 1]
```diff
@@ -255,7 +255,7 @@ export class AmplifyForgotPassword {
 		this.loading = true;
 		try {
 			const { userInput, code, password } = this.forgotPasswordAttrs;
-			const data = await Auth.forgotPasswordSubmit(userInput, code, password);
+			const data = await Auth.forgotPasswordSubmit(userInput.trim(), code, password);
```
Comment (wlee221) [inline packages/amplify-ui-components/src/components/amplify-forgot-password/amplify-forgot-password.tsx:258]: What happens if the `userInput` is empty? Do we get a friendly error message from Cognito? If not, we might need to check for empty string as well.

**Review by wlee221 — COMMENTED** [review]
Hi @dennybiasiolli, thanks for your contribution! Just got one question -- the rest LGTM.

File: packages/amplify-ui-components/src/components/amplify-forgot-password/amplify-forgot-password.tsx (hunk 1 above)
Comment (dennybiasiolli) [reply packages/amplify-ui-components/src/components/amplify-forgot-password/amplify-forgot-password.tsx:258]: Empty `userInput` is already handled [here](https://github.com/aws-amplify/amplify-js/pull/8468/files#diff-c7d24414916a066a072853fb092792f8cb8e8b9aa454680830664cc40b8b63edR219-R221) in the `send` function, and gives a nice `"Username cannot be empty"` message 😉

**Review by eddiekeller — APPROVED** [review]
LGTM, thanks!

File: packages/amplify-ui-components/src/components/amplify-forgot-password/amplify-forgot-password.tsx (hunk 1 above)
Comment (wlee221) [reply packages/amplify-ui-components/src/components/amplify-forgot-password/amplify-forgot-password.tsx:258]: Ah, got it! Thanks for clarifying.

**Review by wlee221 — APPROVED** [review]
LGTM! Let's get this merged in as soon as CI passes.

**Pull request merged**